### ISSUE TITLE
Arrows missing when printing Java 12 SwitchEntry nodes

### ISSUE THREAD

**Issue by ccreeten** [javaparser/javaparser#2636]
I was trying to parse and (re)save some Java source files when I stumbled on this. When feeding a file containing some Java 12 switch expressions, and then regenerating those files (using SourceRoot), the resulting switch entries had colons between case and expression, not arrows. I created a minimal test to verify it:
```java
SwitchExpr switchExpr = parseExpression("switch (k) {\n" +
        "        case 1 -> 2;\n" +
        "    }");
```
When printing this expression (with default toString, which uses the PrettyPrinter), I get the following output:
```java
switch(k) {
    case 1:
        2;
}
```
Which results in a syntax error. I took a quick peek and it seems like an omission in the PrettyPrintVisitor? Would it be enough to add a case for non statement group entries, printing the arrows?

**Comment by edefazio** [comment]
yes, I can verify this is a problem...
we need to correct the `PrettyPrintVisitor` need to inspect `SwitchEntry` types that are EXPRESSION (and print the ->) verses the STATEMENT_GROUP types (and print the :)  NOTE: I dont think we can always use non-statement group entities, because `SwitchStmt`s wont allow the -> I dont believe (I think it's only for `SwitchExpr`s)

I'm still looking into use cases 
i.e. when we have a code block:
```java
case 1 -> { 
     System.out.println(1);
     yield 'a';
}
```
or when we have multiple cases per return:
```java
case 2, case 3 -> 'a'
```
(but in the interum you could probably update a custom `PrettyPrintVisitor` to make it work

thanks

**Comment by edefazio** [comment]
Just leaving a note to myself :
using this as a baseline:
https://blog.codefx.org/java/switch-expressions/

`SwitchExpr` s are allowed to have either ':' or '->' separators

the ':' is used to denote that fall-though can occur, and (internally) can use the 'yield' keyword/ `YieldStmt` as apposed to a 'return'/ `ReturnStmt`

```java
//this is a SwitchExpr using ":" (fall-through) but yield fixes it
SwitchExpr fallThrough = Expressions.switchEx("switch (k) {\n" +
                "         default : yield 3;\n" +
                "         case 1, 2 : yield 2;\n" +
                "    }");
```

also, seems like we can mix/match explicit cases and comma separated cases:
```java
SwitchExpr fallThrough2 = Expressions.switchEx("switch (k) {\n" +
                "         case 1: case 2 : yield 2;\n" + //individual explicit cases
                "         case 3, 4: yield 2;\n" + // cases comma separated list
                "         default : yield 3;\n" +
                "    }");
```


the '->' is used as a short form to mean return the result of this expression, and NO FALL THROUGH so you don't need to  have a 'return'/ `ReturnStmt`

```java
SwitchExpr se4 = Expressions.switchEx("switch (k) {\n" +
                "         default -> 3;\n" +
                "         case 1, 2 -> 2;\n" +
                "    }");
```
each of which are valid, but I have to find out if we capture (within the `SwitchEntry` of the AST (or can derive) which of these to print when we write it out.)

**Comment by matozoid** [comment]
@edefazio - take a look at the Javadoc for the nodes, it's pretty thorough!

**Comment by edefazio** [comment]
yeah, I'm looking at the source now... but it doesnt seem to clarify or model the difference in semantics between ":" and "->", so when we have to print things out, the best (hack) I can come up with is to read from the `TokenRange` and determine if the `TokenRange` has a `->` or a `:` because the `SwitchExpr.Type` types dont model the difference. 

at first I thought it'd be easy... just look at your parent and the Type to detemine which to use (':' or '->' but turns out `SwitchExpr` can use either ':' or '->' so, at the moment the underlying nodes dont model this & we have to figure out another way to determine whether the code is 

```java
case a -> 'a'; //yield expression
case a -> throw new RuntimeException();
case a : yield 'a'; //explicit Yield Statement
case a : throw new RuntimeException();
case a-> { 3+4; throw RuntimeException()} //yield block stmt
```

the types cover whether its:
```java
    public enum Type {

        STATEMENT_GROUP, EXPRESSION, BLOCK, THROWS_STATEMENT
    }
```

but you cant infer (from this alone & the parent type) whether it's printed with a ':' or a '->'

**Comment by matozoid** [comment]
A statement group is `:`, the rest is `->` according to the docs and the grammar.

**Comment by ccreeten** [comment]
> NOTE: I dont think we can always use non-statement group entities, because `SwitchStmt`s wont allow the -> I dont believe (I think it's only for `SwitchExpr`s)

I based it on the SwitchEntry JavaDoc, which just seems to be based on the spec, as matozoid mentions. In any case, it fixes everything for me locally :D

Perhaps the yield case could be added there as well, however.

**Comment by jlerbsc** [comment]
It appears that this case has been fixed because the result is now 

    switch(k) {
        case 1 ->
            2;
    }
